### DISCUSSION TITLE
implement to control fan with special IR controller

### DISCUSSION THREAD

**Discussion by ChinaXing** [RavenSystem/esp-homekit-devices#2087] [MEPLHAA Scripts]
I use esp8266+IR to control an fan, the fan has following control buttons in its IR-controller

1. turn on/off
2. increment speed
3. decrement speed
4. swift left - right
5. swift up-down

use the SerivceType of fan, but control speed is not easy to implement.

I use the wild action, but it seems when I turn on the fan in homekit, the service action "0" will not be triggered ( because the IR light is not sent)

so, how to fix this, and is there other advice ?
**Detailed steps to reproduce it**
Just use you document

**MEPLHAA Script**
```
{
 "c":{"r":[{"n":0, "s":115200, "p":0, "b":1}]},
 "a":[
  {
    "t": 65,
    "0": {"u":[{"n":0,"v":"680800FF12001116"}]},
    "1": {"u":[{"n":0,"v":"680800FF12001116"}]},
    "s": 32,
    "y0": [
      {"v": 20, "0": {"u":[{"n":0,"v":"680800FF12021316"}]}},
      {"v": 40, "0": {"u":[{"n":0,"v":"680800FF12021316"}]}},
      {"v": 70, "0": {"u":[{"n":0,"v":"680800FF12011216"}]}},
      {"v": 90, "0": {"u":[{"n":0,"v":"680800FF12011216"}]}}
    ],
    "es": [
      {
        "t":1,
        "0": {"u":[{"n":0,"v":"680800FF12041516"}]},
        "1": {"u":[{"n":0,"v":"680800FF12041516"}]}
      },
      {
        "t":1,
        "0": {"u":[{"n":0,"v":"680800FF12031416"}]},
        "1": {"u":[{"n":0,"v":"680800FF12031416"}]}
      }
    ]
  }
  ]
}
```
**Used device**
esp8266-01s
**HAA version**
12, the newest
**iOS/iPadOS version**
newest iphone
**Logs**
A .txt file with logs. Read [how to capture logs from your device](https://github.com/RavenSystem/esp-homekit-devices/wiki/Build-instructions#capture-logs)

**Screen-shots or videos showing issue**
none

**Comment by RavenSystem** [comment]
But what is the bug? I will move it to discussions.
